lib/DXIL/DxilOperations.cpp

```diff
@@ -3564,8 +3564,18 @@ void OP::UpdateCache(OpCodeClass opClass, Type *Ty, llvm::Function *F) {
 Function *OP::GetOpFunc(OpCode opCode, Type *pOverloadType) {
   if (opCode == OpCode::NumOpCodes)
     return nullptr;
-  if (!IsOverloadLegal(opCode, pOverloadType))
+  if (!pOverloadType)
     return nullptr;
+  // Remove this assert on illegal overload for now.
+  // Illegal overloads are generated and eliminated by DXIL op constant
+  // evaluation for a number of cases where a double overload of an HL intrinsic
+  // that otherwise does not support double is used for literal values, when
+  // there is no constant evaluation for the intrinsic in CodeGen.
+  // Illegal overloads of DXIL intrinsics may survive through to final DXIL,
+  // but these will be caught by the validator, and this is not a regression.
+  // DXASSERT(IsOverloadLegal(opCode, pOverloadType),
+  //      "otherwise the caller requested illegal operation overload (eg HLSL "
+  //      "function with unsupported types for mapped intrinsic function)");
```

Comment:
Since we've closed the issues that would have uncommented this, I think we just need to remove the commented code entirely since we're not fixing it.